File: backend/sabor-da-familia/src/main/java/br/com/puc/saborfamilia/service/perfil/dto/response/PerfilResponse.java
package br.com.puc.saborfamilia.service.perfil.dto.response;

import br.com.puc.saborfamilia.database.entity.PerfilEntity;
import br.com.puc.saborfamilia.service.personalizacao.dto.response.PersonalizacaoResumoResponse;
import br.com.puc.saborfamilia.service.restricao.dto.RestricaoAlimentarResumoResponse;
import com.fasterxml.jackson.annotation.JsonInclude;
import java.time.LocalDateTime;
import java.util.List;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record PerfilResponse(
  Long id,
  Long usuarioId,
  Boolean proprioPerfil,
  Boolean seguindoPerfil,
  DetalhesPerfil detalhes,
  EstatisticasPerfil estatisticas,
  List<RestricaoAlimentarResumoResponse> restricoesAlimentares,
  List<PersonalizacaoResumoResponse> personalizacao,
  LocalDateTime dataCadastro
) {

  public static PerfilResponse fromEntity(
    PerfilEntity perfil,
    Boolean proprioPerfil,
    Boolean seguindoPerfil,
    long seguidores,
    long seguindo,
    List<RestricaoAlimentarResumoResponse> restricoes,
    List<PersonalizacaoResumoResponse> personalizacao
  ) {

    return new PerfilResponse(
      perfil.getId(),
      perfil.getUsuarioId(),
      proprioPerfil,
      seguindoPerfil,
      new DetalhesPerfil(
        perfil.getNome(),
        perfil.getEmail(),
        perfil.getBio(),
        perfil.getFotoPerfilUrl()
      ),
      new EstatisticasPerfil(
        seguidores,
        seguindo
      ),
      restricoes,
      personalizacao,
      perfil.getDataCadastro()
    );
  }
}

File: frontend/src/features/sabor_familia/dto/perfil/response/PerfilResponse.ts

import type { DetalhesPerfil } from "./DetalhesPerfil";
import type { EstatisticasPerfil } from "./EstatisticasPerfil";
import type { RestricaoAlimentarResumoResponse } from "./../../restricao/response/RestricaoAlimentarResumoResponse";
import type { PersonalizacaoResumoResponse } from "../../personalizacao/response/PersonalizacaoResumoResponse";

export interface PerfilResponse {
  id: number;
  usuarioId: number;
  proprioPerfil: boolean;
  seguindoPerfil?: boolean;
  detalhes: DetalhesPerfil;
  estatisticas: EstatisticasPerfil;
  restricoesAlimentares: RestricaoAlimentarResumoResponse[];
  personalizacao: PersonalizacaoResumoResponse[];
  dataCadastro: string;
}
